grammar/TypeQL.g4

```diff
@@ -152,16 +161,28 @@ attributes            :   attribute ( ',' attribute )* ;
 attribute             :   HAS label ( VAR_ | predicate )                        // ownership by labeled variable or value
                       |   HAS VAR_ ;                                            // or just value
 
-// ATTRIBUTE VALUATION CONSTRUCTS ==============================================
+// PREDICATE CONSTRUCTS ========================================================
 
-predicate             :   value
+predicate             :   value |   VALVAR_   |
                       |   predicate_equality   predicate_value
                       |   predicate_substring  STRING_
                       ;
 predicate_equality    :   EQ | NEQ | GT | GTE | LT | LTE ;
 predicate_substring   :   CONTAINS | LIKE ;
 
-predicate_value       :   value | VAR_  ;
+predicate_value       :   value | VAR_  | VALVAR_ ;
```

Comment:
Move VALVAR_ to attribute has rule

Reply:
Revisit.
That causes us to either 
* handle it as something that isn't an == predicate OR
* write the rule conclusion contains a ValueVariable as `$x has attr_type == ?v`